programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -2288,7 +2288,7 @@ mod tests {
         // zero len
         let good_data = vec![1u8, 2, 3, 4, 5];
         let data: Vec<u8> = vec![];
-        assert_eq!(0x1 as *const u8, data.as_ptr());
+        assert_eq!(std::ptr::dangling::<u8>(), data.as_ptr());
```

Comment:
Hmm, I didn't notice this one before; I might break this into a separate PR in order to keep this PR focused on the simpler string format changes